hypothesis-python/src/hypothesis/extra/cli.py

```diff
@@ -0,0 +1,136 @@
+    # on the principle that Ghostwriter should 'just work' for novice users.  Note
+    # that we append rather than prepend to the module search path, so this will
+    # never shadow the stdlib or installed packages.
+    sys.path.append(".")
+
+    @click.group(context_settings={"help_option_names": ("-h", "--help")})
+    @click.version_option()
+    def main():
+        pass
+
+    def obj_name(s: str) -> object:
+        """This "type" imports whatever object is named by a dotted string."""
+        try:
+            return __import__(s)
+        except ImportError:
+            pass
+        if "." not in s:
+            modulename, module, funcname = "builtins", builtins, s
+        else:
+            modulename, funcname = s.rsplit(".", 1)
+            try:
+                module = __import__(modulename)
+            except ImportError:
+                raise click.UsageError(
+                    f"Failed to import the {modulename} module for introspection.  "
+                    "Check spelling and your Python import path, or use the Python API?"
+                )
+        try:
+            return getattr(module, funcname)
+        except AttributeError:
+            public_names = [name for name in vars(module) if not name.startswith("_")]
+            matches = get_close_matches(funcname, public_names)
+            raise click.UsageError(
+                f"Found the {modulename!r} module, but it doesn't have a "
+                f"{funcname!r} attribute."
+                + (f"  Closest matches: {matches!r}" if matches else "")
+            )
+
+    @main.command()  # type: ignore  # Click adds the .command attribute
+    @click.argument("func", type=obj_name, required=True, nargs=-1)
+    @click.option("--idempotent", "writer", flag_value="idempotent")
+    @click.option("--equivalent", "writer", flag_value="equivalent")
+    @click.option("--roundtrip", "writer", flag_value="roundtrip")
+    @click.option(
+        "--style",
+        type=click.Choice(["pytest", "unittest"]),
+        default="pytest",
+        help="pytest-style function, or unittest-style method?",
+    )
+    @click.option(
+        "-e",
+        "--except",
+        "except_",
+        type=obj_name,
+        multiple=True,
+        help="dotted name of exception(s) to ignore",
+    )
+    def write(func, writer, except_, style):  # noqa: D301  # \b disables autowrap
+        """`hypothesis write` writes property-based tests for you!
+
+        Type annotations are helpful but not required for our advanced introspection
+        and templating logic.  Try running the examples below to see how it works:
+
+        \b
+            hypothesis write gzip
+            hypothesis write re.compile --except re.error
+            hypothesis write --style=unittest --idempotent sorted
+            hypothesis write --equivalent ast.literal_eval eval
+            hypothesis write --roundtrip json.dumps json.loads
+        """
```

Comment:
Even though this is being made available through a CLI, it would still be useful to document the parameters here. Type-annotations seem warranted as well, unless they are being omitted for a reason.

Reply:
I was not anticipating that this would ever be used except via click, which provides that extra documentation in the help output - and also does argument type conversion. What did you have in mind?

Reply:
Simply a docstring that documents the various parameters. I can imagine people being interested in tinkering with this outside of the cli

Reply:
I'll add a comment directing people to the `hypothesis.extra.ghostwriter` module then; you can't really call a click-decorated function directly - even if you get the `sys.argv`-style list of inputs right, it'll still `raise SystemExit` and quit when it's finished.